spanner/client.go

```diff
@@ -276,16 +302,29 @@ func NewClientWithConfig(ctx context.Context, database string, config ClientConf
 	if config.Compression == gzip.Name {
 		md.Append(requestsCompressionHeader, gzip.Name)
 	}
+
 	// Create a session client.
 	sc := newSessionClient(pool, database, config.UserAgent, sessionLabels, config.DatabaseRole, config.DisableRouteToLeader, md, config.BatchTimeout, config.Logger, config.CallOptions)
 
+	// Create a OpenTelemetry configuration
+	otConfig, err := getOpenTelemetryConfig(config.OpenTelemetryMeterProvider, config.Logger, sc.id, database)
```

Comment:
What happens if a user tries to call `NewClientWithConfig` in multiple goroutines at the same time? I don't think `getOpenTelemetryConfig` will work correctly in that case. And that again is an indication that the name of the method should probably be `getOrCreate...`

Reply:
As mentioned above, this should be called `createOpenTelemetryConfig`.

When `NewClientWithConfig` gets called in multiple goroutines, the behavior would be similar to `newSessionClient`, because `createOpenTelemetryConfig` does not have any shared resource and its functionality is to create a new config object and return it back to client.

Please correct me if I am missing something here.

Reply:
I re-read that piece of code, and I agree with you that it is setup in the correct way, so each client gets its own config, and that there does not seem to be any possible data races or similar.